packages/amplify-graphql-model-transformer/rds-lambda/clients/mysql/MySQLPasswordClient.ts

```diff
@@ -0,0 +1,57 @@
+import Knex from 'knex';
+import { SSMClient, GetParameterCommand } from '@aws-sdk/client-ssm';
+import { MySQLClient } from './MySQLClient';
+
+/**
+ * Postgres client with password connection implementation
+ */
+export class MySQLPasswordClient extends MySQLClient {
+  private clientPromise: Promise<any>;
+  constructor() {
+    super();
+    this.clientPromise = this.getClient();
+    this.clientPromise.then((value) => this.client = value);
+  }
+
+  getClient = async (): Promise<any> => {
+    if (this.client) {
+      return this.client;
+    }
+    if (this.clientPromise) {
+      await this.clientPromise;
+      return this.client;
+    }
+    const passwordClient = this.getSSMClient();
+    return Knex({
+      client: 'mysql',
+      connection: {
+        host: process.env.host,
+        port: Number.parseInt(process.env.port ?? '3306', 10),
+        user: await this.getSSMValue(passwordClient, process.env.username),
+        password: await this.getSSMValue(passwordClient, process.env.password),
+        database: process.env.database,
+      },
+    });
+  }
+
+  private getSSMClient(): SSMClient {
+    return new SSMClient({});
+  }
+
+  private async getSSMValue(client: SSMClient, key: string | undefined): Promise<string> {
+    if (!key) {
+      return '';
```

Comment:
Would prefer to throw here rather than defaulting to prevent unintended DB connections with empty username/passwords.

Reply:
Fixed locally and about to commit